subsys/nrf_security/src/drivers/cracen/cracenpsa/src/pake.c

```diff
@@ -110,56 +43,43 @@ psa_status_t cracen_pake_set_user(cracen_pake_operation_t *operation, const uint
 {
 	psa_status_t status = PSA_ERROR_BAD_STATE;
 
-	switch (operation->alg) {
-	case PSA_ALG_JPAKE:
-		if (IS_ENABLED(PSA_NEED_CRACEN_ECJPAKE_SECP_R1_256)) {
-			status = cracen_jpake_set_user(&operation->cracen_jpake_ctx, user_id,
-						       user_id_len);
-		}
-		break;
-	case PSA_ALG_SRP_6:
-		if (IS_ENABLED(PSA_NEED_CRACEN_SRP_6)) {
-			status = cracen_srp_set_user(&operation->cracen_srp_ctx, user_id,
-						     user_id_len);
-		}
-		break;
-	case PSA_ALG_SPAKE2P:
-		if (IS_ENABLED(PSA_NEED_CRACEN_SPAKE2P)) {
-			status = cracen_spake2p_set_user(&operation->cracen_spake2p_ctx, user_id,
-							 user_id_len);
-		}
-		break;
-	default:
-		(void)user_id;
-		(void)user_id_len;
+	if (PSA_ALG_IS_JPAKE(operation->alg) && IS_ENABLED(PSA_NEED_CRACEN_ECJPAKE_SECP_R1_256)) {
+		status = cracen_jpake_set_user(&operation->cracen_jpake_ctx, user_id, user_id_len);
+	} else if (PSA_ALG_IS_SRP_6(operation->alg) && IS_ENABLED(PSA_NEED_CRACEN_SRP_6)) {
+		status = cracen_srp_set_user(&operation->cracen_srp_ctx, user_id, user_id_len);
+	} else if (PSA_ALG_IS_SPAKE2P(operation->alg) && IS_ENABLED(PSA_NEED_CRACEN_SPAKE2P)) {
+		status = cracen_spake2p_set_user(&operation->cracen_spake2p_ctx, user_id,
+						 user_id_len);
 	}
 
 	return status;
 }
 
 psa_status_t cracen_pake_set_peer(cracen_pake_operation_t *operation, const uint8_t *peer_id,
 				  size_t peer_id_len)
+{
+	if (PSA_ALG_IS_JPAKE(operation->alg) && IS_ENABLED(PSA_NEED_CRACEN_ECJPAKE_SECP_R1_256)) {
+		return cracen_jpake_set_peer(&operation->cracen_jpake_ctx, peer_id, peer_id_len);
+	} else if (PSA_ALG_IS_SPAKE2P(operation->alg) && IS_ENABLED(PSA_NEED_CRACEN_SPAKE2P)) {
+		return cracen_spake2p_set_peer(&operation->cracen_spake2p_ctx, peer_id,
+					       peer_id_len);
+	} else {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+}
+
+psa_status_t cracen_pake_set_context(cracen_pake_operation_t *operation, const uint8_t *context,
+				     size_t context_length)
 {
 	psa_status_t status = PSA_ERROR_BAD_STATE;
```

Comment:
Should this be `BAD_STATE` or rather `NOT_SUPPORTED` as we don't have have 
`else {
		return PSA_ERROR_NOT_SUPPORTED;
	}`   in this function at the end of the if / else cascade

Reply:
These would have failed in cracen_pake_setup, so bad state seem appropriate